crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/stateless/StatelessAssessmentService.java

```diff
@@ -34,97 +31,92 @@ public StatelessAssessmentService(AssessmentCriteriaService assessmentCriteriaSe
-                    .result(fullAssessmentResult.getFullAssessmentResult())
-                    .disposableIncome(fullAssessmentResult.getTotalAnnualDisposableIncome())
-                    .adjustedLivingAllowance(fullAssessmentResult.getAdjustedLivingAllowance())
-                    .totalAnnualAggregatedExpenditure(fullAssessmentResult.getTotalAggregatedExpense())
-                    .eligibilityThreshold(criteriaEntry.getEligibilityThreshold()).build();
-            return new StatelessResult(statelessFullResult, initialAnswer);
-        } else {
-            return new StatelessResult(null, initialAnswer);
-        }
-    }
+        // assessmentStatus has to be set 'COMPLETE' otherwise the return value is null
+        final MeansAssessmentRequestDTO requestDTO = MeansAssessmentRequestDTO
+                .builder()
+                .assessmentStatus(CurrentStatus.COMPLETE)
+                .childWeightings(children)
+                .initTotalAggregatedIncome(totalIncome)
+                .eligibilityCheckRequired(eligibilityCheckRequired)
+                .build();
 
-    private StatelessResult initialOnly(@NotNull Assessment assessment,
-                                        @NotNull Map<AgeRange, Integer> childGroupings,
-                                        @NotNull List<Income> incomes) {
+        final var totalOutgoings = calcOutgoingTotals(criteriaEntry, caseType, outgoings);
+        final var fullAssessmentService = meansAssessmentServiceFactory.getService(AssessmentType.FULL);
+        final var fullAssessmentResult = fullAssessmentService.execute(totalOutgoings, requestDTO, criteriaEntry);
 
-        final var criteriaEntry = assessmentCriteriaService.getAssessmentCriteria(
-                assessment.getAssessmentDate(), assessment.getHasPartner(), false
-        );
-        final var totalIncome = calcIncomeTotals(criteriaEntry, assessment.getCaseType(), incomes);
-        var initialAssessment = initialResult(assessment, totalIncome, childGroupings);
-        return new StatelessResult(null, initialAssessment);
+        return StatelessFullResult.builder()
+                .result(fullAssessmentResult.getFullAssessmentResult())
+                .disposableIncome(fullAssessmentResult.getTotalAnnualDisposableIncome())
+                .adjustedLivingAllowance(fullAssessmentResult.getAdjustedLivingAllowance())
+                .totalAnnualAggregatedExpenditure(fullAssessmentResult.getTotalAggregatedExpense())
+                .eligibilityThreshold(criteriaEntry.getEligibilityThreshold()).build();
     }
 
-    private StatelessInitialResult initialResult(@NotNull Assessment assessment,
-                                                 @NotNull BigDecimal totalIncome,
-                                                 @NotNull Map<AgeRange, Integer> childGroupings) {
+    public StatelessInitialResult initialResult(@NotNull Map<AgeRange, Integer> childGroupings,
```

Comment:
Original comment from @Mhowell494:
> I think we should remove the incomes parameter here and the outgoings one in the fullResult method. Instead, the methods should accept a BigDecimal total. This way, the MeansAssessmentService can call calculateSummariesTotal and then simply pass in the overall total. Then it's just a case of moving the calcIncomeTotals and calcOutgoingTotals to the execute method.

Reply:
The thinking is:
* calculateSummariesTotal is a key part of the business logic for the means test, because the rules change over time to include different income and outgoing categories
* we want the business logic encapsulated inside StatelessAssessmentService (and downstream services)